infra/templates/az-isolated-service-single-region/.env.template

```diff
@@ -0,0 +1,15 @@
+ARM_SUBSCRIPTION_ID="<az-service-principal-subscription-id>"
+ARM_CLIENT_ID="<az-service-principal-client-id>"
+ARM_CLIENT_SECRET="<az-service-principal-auth-secret>"
+ARM_TENANT_ID="<az-service-principal-tenant>"
+ARM_ACCESS_KEY="<remote-state-storage-account-primary-key>"
+TF_VAR_remote_state_account="<tf-remote-state-storage-account-name>"
+TF_VAR_remote_state_container="<tf-remote-state-storage-container-name>"
+TF_VAR_admin_subscription_id="<subscription for hosting admin resources>"
+TF_VAR_app_service_environment_name="<name of app service environment>"
+TF_VAR_app_service_environment_resource_group="<resource group hosting app service environment>"
+GO_VERSION=1.12.5
+TF_VERSION=0.12.2
+BUILD_BUILDID=1
+DATACENTER_LOCATION=eastus2
+TF_WARN_OUTPUT_ERRORS=1
```

Comment:
Wouldn't this mean that we'd have to redefine the `.env` for each template?

Reply:
Yes it does. For example, an AWS template will need different environment variables from an Azure template

Reply:
removed.

Reply:
Concerned about the developer experience considering the scenario we have 20+ cobalt templates for azure where most of those `.env` files will comprise of duplicate environment definitions. An alternate approach would be to support `.env ` overlays in each template folders but still have the root `.env` file provide the default environment.

Reply:
I'm going to delete this file for now since the only additions are `TF_VAR_*` which need not be environment-specific. 

I agree that we should think more about this before landing on a new approach.